Source/DafnyDriver/Legacy/SynchronousCliCompilation.cs

```diff
@@ -629,6 +630,18 @@ private record TargetPaths(string Directory, string Filename) {
     public static async Task<bool> CompileDafnyProgram(Program dafnyProgram, string dafnyProgramName,
                                            ReadOnlyCollection<string> otherFileNames, bool invokeCompiler) {
 
+      //This recursive call is to produce a doo artifact for every backend.
+      if (dafnyProgram.Options.Backend is not LibraryBackend && dafnyProgram.Options.UsingNewCli) {
+        var b = dafnyProgram.Options.Backend;
+        var lib = new LibraryBackend(dafnyProgram.Options);
+        lib.ProgramAfterParsing = dafnyProgram.Options.Backend.ProgramAfterParsing;
+        dafnyProgram.Options.Backend = lib;
+        var runAfterCompile = dafnyProgram.Options.RunAfterCompile;
+        dafnyProgram.Options.RunAfterCompile = false;
+        await CompileDafnyProgram(dafnyProgram, dafnyProgramName, otherFileNames, true);
+        dafnyProgram.Options.Backend = b;
+        dafnyProgram.Options.RunAfterCompile = runAfterCompile;
+      }
```

Comment:
I wonder if it would be cleaner to have the common logic for backends emit doo files, and then the LibraryBackend just becomes a no-op that doesn't emit anything else on top of that.

Reply:
Made this whole thing optional using the `--generate-doo` flag.